response_operations_ui/views/passwords.py

```diff
@@ -0,0 +1,141 @@
+        return render_template('reset-password-expired.html', token=token)
+    except (BadSignature, BadData):
+        logger.warning('Invalid token sent to Response Operations password reset', token=token)
+        return render_template('reset-password-expired.html', token=token)
+
+    template_data = {
+        "error": {
+            "type": form_errors
+        },
+        'token': token
+    }
+    return render_template('reset-password.html', form=form, data=template_data)
+
+
+@passwords_bp.route('/reset-password/<token>', methods=['POST'])
+def post_reset_password(token):
+    form = ResetPasswordForm(request.form)
+
+    if not form.validate():
+        return get_reset_password(token, form_errors=form.errors)
+
+    password = request.form.get('password')
+
+    try:
+        duration = app.config['EMAIL_TOKEN_EXPIRY']
+        email = token_decoder.decode_email_token(token, duration)
+    except SignatureExpired:
+        logger.warning('Token expired for Response Operations password reset', token=token)
+        return render_template('reset-password-expired.html', token=token)
+    except (BadSignature, BadData):
+        logger.warning('Invalid token sent to Response Operations password reset', token=token)
+        return render_template('reset-password-expired.html', token=token)
+
+    if (uaa_controller.change_user_password(email, password)):
+        logger.info('Successfully changed user password', token=token)
+        return redirect(url_for('passwords_bp.reset_password_confirmation'))
+
+    logger.warning('Error changing password in UAA', token=token)
+    return render_template('reset-password-error.html')
+
+
+@passwords_bp.route('/reset-password/confirmation', methods=['GET'])
+def reset_password_confirmation():
+    return render_template('passwords/reset-password-confirmation.html')
+
+
+@passwords_bp.route('/resend-password-email-expired-token/<token>', methods=['GET'])
+def resend_password_email_expired_token(token):
+    email = token_decoder.decode_email_token(token)
+    return send_password_change_email(email)
+
+
+def send_password_change_email(email):
+    first_name = uaa_controller.get_first_name_by_email(email)
+    if first_name != "":
+        internal_url = app.config['INTERNAL_WEBSITE_URL']
+        verification_url = f'{internal_url}/passwords/reset-password/{token_decoder.generate_email_token(email)}'
+
+        personalisation = {
+            'RESET_PASSWORD_URL': verification_url,
+            'FIRST_NAME': first_name
+        }
+
+        try:
+            NotifyController(app.config).request_to_notify(email=email,
+                                                           template_name='confirm_password_change',
+                                                           personalisation=personalisation)
+        except NotifyError:
+            logger.error('Error sending password change request email to Notify Gateway')
```

Comment:
Log the message we get form the error here.  Nothing else will log that for you and we'll need it for debugging